client/client.go

```diff
@@ -218,14 +234,41 @@ func (c Client) RoleTargetsPath(roleName string, snapshotMeta data.Files, consis
 		}
 		roleName = path.Join(
 			dir,
-			fmt.Sprintf("%s.%s.json", roleMeta.Hashes["sha256"], roleName),
+			fmt.Sprintf("%s.%s.json", hasheSha256, roleName),
 		)
 	}
 	return roleName, nil
 }
 
-func (c Client) TargetMeta(path string) *data.FileMeta {
-	return c.local.FindTarget(path)
+// TargetMeta ensures the repo is up to date, downloading the minimum
+// necessary metadata files
+func (c Client) TargetMeta(path string) (*data.FileMeta, error) {
+	c.Update()
+	var meta *data.FileMeta
+
+	// FIFO list of targets delegations to inspect for target
+	roles := []string{data.ValidRoles["targets"]}
+	for ; len(roles) > 0; role := roles[0] {
+		roles = roles[1:]
+		// Download the target role file if necessary
+		err = c.downloadTargets(role)
+		if err != nil {
+			// as long as we find a valid target somewhere we're happy.
+			// continue and search other delegated roles if any
+			continue
+		}
+
+		meta = c.local.TargetMeta(role, path)
+		if meta != nil {
+			// we found the target!
+			break
+		}
+		delegations := c.local.TargetDelegations(role, path, pathHex)
+		for _, d := range delegations {
+			roles = append(roles, d.Name)
+		}
+	}
+	return meta, nil
```

Comment:
Yes, that was the intention. It's very easy for users to request things (typo, intentionally, etc...) that aren't in the targets. In considering that normal operation, it's inappropriate to raise an error when something isn't found (an error should indicate something wrong or abnormal). Returning nil let's the caller decide how to handle an empty response.

N.B. the crowdmob goamz library took a similar decision regarding lookups in S3. Originally they returned a 404 as an error, then eventually switched it so a 404 was a regular response with a nil error. It was left up to the caller to determine what an empty response means.